File: FrontEnd/src/app/DTOs/user/user.DTO.ts
import {
    IsString,
    IsNotEmpty,
    IsEnum,
    IsDateString
} from 'class-validator';
import { Gender } from '../../Enums/gender.enum';

export class UserDTO {
    @IsString()
    @IsNotEmpty()
    phone_number!: string;

    @IsString()
    @IsNotEmpty()
    full_name!: string;

    @IsEnum(Gender)
    gender!: Gender;

    @IsString()
    @IsNotEmpty()
    address!: string;

    @IsDateString()
    date_of_birth!: string;

    constructor(data: any) {
        this.phone_number = data.phone_number;
        this.full_name = data.full_name;
        this.gender = data.gender;
        this.address = data.hometown;
        this.date_of_birth = data.birthDate;
    }
}

File: BackEnd/src/main/java/com/example/BackEnd/DTO/UserDTO.java
package com.example.BackEnd.DTO;

import com.example.BackEnd.Enum.Gender;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import lombok.*;

import java.time.LocalDate;

@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class UserDTO {
    @JsonProperty("email")
    @Email(message = "Invalid email format")
    private String email;

    @JsonProperty("phone_number")
    private String phoneNumber;

    @NotBlank(message = "Password can not be blank")
    private String password;

    @JsonProperty("retype_password")
    private String retypePassword;

    @JsonProperty("full_name")
    private String fullName;

    private String address;

    @Enumerated(EnumType.STRING)
    private Gender gender;

    @JsonProperty("date_of_birth")
    private LocalDate dateOfBirth;

    @JsonProperty("role_id")
    private Integer roleId;
}